errors.go

```diff
@@ -70,13 +70,27 @@ func newCellNameToCoordinatesError(cell string, err error) error {
 	return fmt.Errorf("cannot convert cell %q to coordinates: %v", cell, err)
 }
 
+// newColumnNumberError defined the error message on receiving invalid column
+// number.
+func newColumnNumberError(count int) error {
+	var detail string
+	if count < MinColumns {
+		detail = fmt.Sprintf("%d is less than minimum limit of 1 columns", count)
+	} else if count > TotalColumns {
+		detail = fmt.Sprintf("%d exceeds maximum limit of %d columns", count, TotalColumns)
+	} else {
+		detail = "internal error: incorrect error usage"
+	}
+	return fmt.Errorf("%w: %s", ErrColumnNumber, detail)
+}
+
 var (
 	// ErrStreamSetColWidth defined the error message on set column width in
 	// stream writing mode.
 	ErrStreamSetColWidth = errors.New("must call the SetColWidth function before the SetRow function")
 	// ErrColumnNumber defined the error message on receive an invalid column
 	// number.
-	ErrColumnNumber = errors.New("column number exceeds maximum limit")
+	ErrColumnNumber = errors.New("invalid column number")
```

Comment:
I suggest removing this exported error variable and using the `newColumnNumberError` instead of it.